### PR TITLE
Add type declarations

### LINKED ISSUE #6: TypeScript support
It seems like there's no type definition file available in the package. Is this in the roadmap?

### PR DESCRIPTION
**Pull request by franciscohanna92**
Based on the initial work made by @sirkrisp in [this comment](https://github.com/pushy/pushy-node/issues/6#issuecomment-1035333625)

closes #6

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    }
+
+    interface DeviceInfo {
+        /**  Metadata object for the device. */
+        device: {
+            /**  The device's registration date (unix timestamp). */
+            date: number;
+
+            /**  The device platform string identifier. */
+            platform: 'android' | 'ios' | 'web';
```
Comment (franciscohanna92) [inline index.d.ts:137]: @pushy Are this values correct? It's not specified on the docs

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+         * @param recipient one or array of device token OR one or array of topics
+         * @param options Extra options for the notification
+         *
+         * @returns pushId
+         */
+        sendPushNotification(
+            data: unknown,
+            recipient: string | Array<string>,
+            options?: SendPushNotificationOptions
+        ): Promise<string>;
```
Comment (franciscohanna92) [inline index.d.ts]: Should we type the optional `callback` parameter? 
```suggestion
        sendPushNotification(
            data: unknown,
            recipient: string | Array<string>,
            options?: SendPushNotificationOptions,
            callback?: (error: Error | null, pushId: string) => void
        ): Promise<string>;
```

File: index.d.ts (hunk 1 above)
Comment (pushy) [reply index.d.ts:137]: Nice catch. There are two more platforms: `electron` and `python`.

File: index.d.ts (hunk 2 above)
Comment (pushy) [reply index.d.ts]: Yes, we think this would be preferable.

**Comment by pushy** [comment]
This is really well done. Thanks so much for your work @franciscohanna92 and @sirkrisp! 

Your PR has been merged and published to npm in version `2.0.14`. 🎉

**Pull request merged**